modules/gojira-core/src/main/java/com/flipkart/gojira/external/ManagedImpl.java

```diff
@@ -39,6 +40,23 @@ public void setup() throws SetupException {
     RmqManager.RMQ_MANAGER.setup();
   }
 
+  @Override
```

Comment:
Usually we expect connections to be established during bootstrap itself. Here since we are doing it via APIs, we may have to add a lock per case and execute so that they are done serially. Can u add a lock/unlock using ReentrantLock to make this thread-safe?